sympy/physics/control/lti.py

```diff
@@ -0,0 +1,186 @@
+from sympy import Basic, Mul, degree, Symbol, expand, cancel, Expr
+from sympy.core.numbers import Integer, Float
+from sympy.core.sympify import sympify, _sympify
+
+__all__ = ['TransferFunction',]
+
+
+class TransferFunction(Basic):
+
+    def __new__(cls, num, den, var):
+        num, den = _sympify(num), _sympify(den)
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Variable input must be a Symbol.")
+        if den == 0:
+            raise ValueError("TransferFunction can't have a zero denominator.")
+
+        if (((isinstance(num, Expr) and num.has(Symbol)) or num.is_number) and
+            ((isinstance(den, Expr) and den.has(Symbol)) or den.is_number)):
+                obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+                obj._num = num
+                obj._den = den
+                obj._var = var
+                return obj
+        else:
+            raise ValueError("Unsupported type for numerator or denominator of TransferFunction.")
+
+    @property
+    def num(self):
+        return self._num
+
+    @property
+    def den(self):
+        return self._den
+
+    @property
+    def var(self):
+        return self._var
+
+    @property
+    def bound_symbols(self):
+        return [self.var]
```

Comment:
I haven't seen any unit tests for this yet. What is it's purpose?

Reply:
Please see https://github.com/sympy/sympy/pull/19390#issuecomment-640556483
Yes, I will add tests for this property!

Reply:
As it is written it seems redundant to var. It don't see any need for it without more explanation.

Reply:
If it seems redundant, I can remove it. It is only needed in `_eval_subs` which can also work without this property.

Reply:
If this is a standard method that sympy objects have and expect, then you can include it, otherwise I'd remove.

Reply:
Bound symbols is a standard method for Basic subclasses although it is only implemented on those class that need it. The Basic methods `as_dummy` and `dummy_eq` depend on `bound_symbols` and `_eval_subs` to work. The `as_dummy` method returns an object replacing all bound symbols with canonical symbols and `dummy_eq` uses this to compare to objects modulo bound symbols e.g.:
```julia
In [21]: Integral(x, (x, y, z)).free_symbols                                                                                                   
Out[21]: {y, z}

In [22]: Integral(x, (x, y, z)).bound_symbols                                                                                                  
Out[22]: [x]

In [23]: Integral(x, (x, y, z)).as_dummy()                                                                                                     
Out[23]: 
z         
⌠         
⎮ _0 d(_0)
⌡         
y         

In [24]: Integral(x, (x, y, z)).dummy_eq(Integral(t, (t, y, z)))                                                                               
Out[24]: True

In [25]: Integral(x, (x, y, z)) == Integral(t, (t, y, z))                                                                                      
Out[25]: False
```
It is not possible to infer the bound symbols from the free symbols because a symbol can be both free and bound. It is expected that `_eval_subs` can take care of ensuring that bound symbols are not substituted in a call to `subs`. That is how `as_dummy` works:
```julia
In [27]: Integral(x, (x, 0, x)).free_symbols                                                                                                   
Out[27]: {x}

In [28]: Integral(x, (x, 0, x)).bound_symbols                                                                                                  
Out[28]: [x]

In [29]: Integral(x, (x, 0, x)).subs(x, y).xreplace({x:Symbol('_0')}).subs(y, x)                                                               
Out[29]: 
x         
⌠         
⎮ _0 d(_0)
⌡         
0         

In [30]: Integral(x, (x, 0, x)).as_dummy()                                                                                                     
Out[30]: 
x         
⌠         
⎮ _0 d(_0)
⌡         
0 
```